### PR TITLE
Docs  add tutorials

### PR DESCRIPTION
**Pull request by jasonwilliams14**
This is initial PR request to add tutorials for NGINX Ingress controller. Initial run to make sure all looks good.

### CODE REVIEW

File: docs/content/tutorials/daemonset-different-port.md [hunk 1]
```diff
@@ -0,0 +1,87 @@
+##  Configuring a NIC `daemonsets` to listen with non-default different ports
```
Comment (brianehlert) [inline docs/content/tutorials/daemonset-different-port.md:1]: Configuring a daemonset to listen on ports other than 80 and 443

File: docs/content/tutorials/nginx-ingress-istio.md [hunk 2]
```diff
@@ -0,0 +1,282 @@
...
+nginx-ingress          Active   27h   istio-injection=enabled
+```
+
+After we have setup and configured Istio, we can then deploy NGINX Plus Ingress as well as our applications that will be part of the service mesh. Istio will now inject sidecar proxies based upon how we have configured Istio (namespace configuration).     
+Now, our deployment will look like the following (with Envoy sidecar proxies).
+
+![nginx-ingress](../../../docs/static/img/nic_istio_small.png)
+
+
+We can now see that after configuring Istio with the necessary pieces needed to install a sidecar proxy into the same pod as NGINX ingress, after we deploy NGINX ingress, there are two containers in the same pod for Nginx Ingress:  one is NGINX-ingress, the other is the istio sidecar proxy.
```
Comment (Jcahilltorre) [inline docs/content/tutorials/nginx-ingress-istio.md:149]:
```suggestion
After configuring Istio with the pieces needed to install a sidecar proxy into the same pod as NGINX Ingress, and deploying NGINX Ingress, there are two containers in the same pod for NGINX Ingress:  one is NGINX-ingress, the other is the istio sidecar proxy.

```

File: docs/content/tutorials/nginx-ingress-istio.md [hunk 3]
```diff
@@ -0,0 +1,282 @@
...
+    action:
+      proxy:
+        upstream: coffee
+        requestHeaders:
+          set:
+          - name: Host
+            value: coffee-svc.nginx-ingress.svc.cluster.local
+```
+
+With our new Host header control in v1.11, when VirtualServer is configured with `requestHeaders`, the value specified will be used and `proxy_set_header $host` will NOT be used. By enabling `use-cluster-ip` to **true**, NGINX will forward requests to the service IP. In our above example, that would be `tea-svc` and `coffee-svc`.
```
Comment (Jcahilltorre) [inline docs/content/tutorials/nginx-ingress-istio.md:205]:
```suggestion
With our new Host header control in v1.11, when VirtualServer is configured with `requestHeaders`, the value specified will be used, and `proxy_set_header $host` will NOT be used. By enabling `use-cluster-ip` to **true**, NGINX will forward requests to the service IP. In our above example, that would be `tea-svc` and `coffee-svc`.

```

File: docs/content/tutorials/nginx-ingress-istio.md [hunk 4]
```diff
@@ -0,0 +1,282 @@
...
+
+Here is the output of `nginx -T` to show our upstreams and proxy_set_header values 
+The server in the upstream is the IP address of the service for that given application)
```
Comment (Jcahilltorre) [inline docs/content/tutorials/nginx-ingress-istio.md:208]:
```suggestion
(The server in the upstream is the IP address of the service for that given application)

```
Missing opening parenthesis?

File: docs/content/tutorials/nginx-ingress-istio.md [hunk 5]
```diff
@@ -0,0 +1,282 @@
...
+Content-Type: text/plain
+Content-Length: 159
+Connection: keep-alive
+expires: Thu, 25 Mar 2021 18:47:20 GMT
+cache-control: no-cache
+x-envoy-upstream-service-time: 0
+x-envoy-decorator-operation: coffee-svc.nginx-ingress.svc.cluster.local:80/*
+```
+
+We can see in the above output, our curl request is sent and received by NGINX Ingress. We can see that the envoy sidecar proxy then sends the request to the service IP to the application (coffee), with the full request being complete and correct. Now we have a full working NGINX+ Ingress with Istio as the sidecar proxies are deployed.
```
Comment (Jcahilltorre) [inline docs/content/tutorials/nginx-ingress-istio.md:272]:
```suggestion
In the above output, we can see that our curl request is sent and received by NGINX Ingress. We can see that the envoy sidecar proxy then sends the request to the service IP to the application (coffee), with the full request being complete and correct. Now we have a full working NGINX+ Ingress with Istio as the sidecar proxies are deployed.

```

File: docs/content/tutorials/tcp-udp-load-balancing.md [hunk 6]
```diff
@@ -0,0 +1,49 @@
+---
+title: "How to Configure Load Balancing for Layer 4 Services"
```
Comment (Jcahilltorre) [inline docs/content/tutorials/tcp-udp-load-balancing.md:2]: This file also seems to be the template without modifications. I guess it is a placeholder

File: docs/content/tutorials/daemonset-different-port.md [hunk 7]
```diff
@@ -0,0 +1,87 @@
...
+
+*NOTE:* Since this example includes a second NGINX Ingress deployment, you need to create a second `ingressClass`, one for each deployment.
+Make sure you change `ingressClass` in the `values.yaml` when defining your second deployment.
+
+Then you can use helm to deploy NIC.
+
+<file is modified locally - give me a HELM command to be complete>
+
+After deploying with Helm you should see two NIC deployments.  Both running as a daemonset, with one pod on each node.
+In my lab this looked like:
```
Comment (Jcahilltorre) [inline docs/content/tutorials/daemonset-different-port.md:48]:
```suggestion
In our lab, the deployments look like this:
```

**Review by Jcahilltorre — COMMENTED** [review]
Changed subject from `I` to `We`. There was a mix of both.

File: docs/content/tutorials/daemonset-different-port.md [hunk 8]
```diff
@@ -0,0 +1,87 @@
...
+Controlled By:  DaemonSet/nginx-ingress-nginx-ingress
+Containers:
+  nginx-ingress-nginx-ingress:
+    Container ID:  containerd://547bb21d418ad01c8f8adff67e62d831fd0f615e52d336aa07fee61c0aa1bbc8
+    Image:         nginx/nginx-ingress:2.1.0
+    Image ID:      docker.io/nginx/nginx-ingress@sha256:b94a2ff0fbd36fa7c2b130226475e0cf924dd6a7e038cd20be74149875e3c18a
+    Ports:         8080/TCP, 8443/TCP, 9113/TCP, 8081/TCP
+    Host Ports:    0/TCP, 0/TCP, 0/TCP, 0/TCP
+```
+As you can see in the above output, is my **second** deployment of NGINX Ingress Controller as a Daemonset. It is now running on ports 8080 and 8443, different from my first deployment, which was running on ports 80 and 443.
```
[the remark this answers is not in the document]
Comment (jasonwilliams14) [reply docs/content/tutorials/daemonset-different-port.md:80]: I think this is fine. It is focused on the Kubernetes terminology. The idea is to try and walk through as specifically as we can.

File: docs/content/tutorials/nginx-ingress-istio.md [hunk 9]
```diff
@@ -0,0 +1,282 @@
...
+![nginx-ingress](../../../docs/static/img/nginx_plain.png)
+
+
+When deploying NGINX Ingress controller with Istio, you must deploy the envoy sidecar proxy into the NGINX Ingress controller pod.   
+The image below is what NGINX Ingress and Istio deployment looks like:
+
+![nginx-ingress](../../../docs/static/img/nginx-envoy.png)
+
+
+By default for NGINX Ingress Controller, we populate the upstream server addresses with the endpoint IPs of the pods. NGINX Ingress Controller 1.11 release now supports the ability to configure NGINX Ingress CRDs (virtualServer/virtualServerRoute)to use the `service/cluster IP`. Using this flag,  NGINX Ingress will generate the .conf with the `service/cluster IP` of the service in the `upstreams/servers` section, instead of pod endpoint IPs. To enable NGINX Ingress to route to the `Service IP`, we are going to use a new feature released in 1.11; [use-cluster-ip](https://docs.nginx.com/nginx-ingress-controller/configuration/virtualserver-and-virtualserverroute-resources/#upstream).
```
[the remark this answers is not in the document]
Comment (jasonwilliams14) [reply docs/content/tutorials/nginx-ingress-istio.md:54]: This is bit of a difficult sentence. My preference would be:
```
By default, NGINX Ingress controller populates the `upstream` `servers` with the IP address of the pod endpoint IP addresses. NGINX Ingress controller release 1.11 now supports the ability to populate the `upstream` `servers` with the `cluster-ip` for a given kubernetes service. This is accomplished using our `custom resource definitions` (virtualServer/virtualServerRoute).

**Comment by pleshakov** [comment]
Hi @jasonwilliams14 @brianehlert 

I think it's better to create an enhancement for this https://github.com/nginxinc/kubernetes-ingress/blob/be4d69eb19cbeb942a267ec298245a0c6adbe293/docs/content/tutorials/daemonset-different-port.md  rather than having this tutorial that updates the helm templates. Asking users to update the template is problematic: they will need to maintain a forked version of KIC helm chart.

To support that "natively", we can add the following two parameters to the helm chart:

Parameter | Description | Default
--- | --- | ---
`controller.ports.http.hostPort` | The HTTP port on the host that will be mapped to the HTTP port of the Ingress Controller container. Only applicapble when `controller.kind` is `daemonset` | `80`
`controller.ports.https.hostPort` | The HTTPS port on the host that will be mapped to the HTTP port of the Ingress Controller container. Only applicable when `controller.kind` is `daemonset` | `443`

This is fairly easy to implement

**Pull request closed without merging**